Include/exports.h

```diff
@@ -0,0 +1,20 @@
+#ifndef Py_EXPORTS_H
+#define Py_EXPORTS_H
+
+#if defined(_WIN32) || defined(__CYGWIN__)
+    #define Py_IMPORTED_SYMBOL __declspec(dllimport)
+    #define Py_EXPORTED_SYMBOL __declspec(dllexport)
+    #define Py_LOCAL_SYMBOL
+#else
+    #if __GNUC__ >= 4
```

Comment:
clang also supports this attribute. 

Using `#if __has_attribute(visibility)` should work with both clang and gcc without hardcoding compiler names. This does require GCC 5 or later though.

Reply:
Some versions of e.g. Ubuntu and CentOS are still on GCC 4.8. I'll try to generalise it.